### PR TITLE
Add debugging information

### PR DESCRIPTION
**Pull request by chanmosq**
Signed-off-by: chanmosq <75444793+chanmosq@users.noreply.github.com>

## Change summary

Add information about debugging and crash reporting.

### Submission Checklist:

* [ ] **Descriptive active voice** - Do descriptive sentences have a clear *subject* and *action verb*?
* [ ] **Answer the question at hand** - Does the documentation answer a *what*, *why*, *how*, or *where* type of question?
* [ ] **Consistency** - Does the content consistently follow the [Style Guide](https://o3de.org/docs/contributing/to-docs/style-guide/quick-reference)?
* [ ] **Help the user** - Does the documentation show the user something *meaningful*?

### CODE REVIEW

File: content/docs/user-guide/programming/debugging/crash-dumps.md [hunk 1]
```diff
@@ -1,5 +1,27 @@
...
+
+## Dump file creation
+
+By default, O3DE creates a dump file that includes the stack trace and some variable information. This is set via the `sys_dump_type` [console variable (CVAR)](/docs/user-guide/appendix/cvars/).
+
+If you want to configure the type of crash log information that's generated, set the `sys_dump_type` CVAR to one of the following values:
+- `0` -- Disable crash reporting.
+- `1` -- Generate a stack trace on crash.
+- `2` -- (Default) Generate a stack trace and limited variable information.
+- `3` -- Generate a Full crash log of the stack trace, and all variable and information.
```
Comment (willihay) [inline content/docs/user-guide/programming/debugging/crash-dumps.md:17]:
```suggestion
- `3` -- Generate a full crash log of the stack trace, and all variable information.
```

**Review by willihay — APPROVED** [review]

**Comment by willihay** [comment]
Good info! You might want to remove the link to the GHI since it will only be partially fixed with this PR, otherwise this PR will close the (partially fixed) issue when merged.

**Review by phuchau1989 — APPROVED** [review]

**Review by cgalvan — APPROVED** [review]

File: content/docs/user-guide/programming/debugging/crash-dumps.md [hunk 2]
```diff
@@ -1,5 +1,27 @@
 ---
-title: "Crash Dumps"
-date: 2021-03-02T00:23:53-05:00
+title: Crash Reporting with Dump Files
+linktitle: Crash Reporting
+description: When you encounter a crash in Open 3D Engine (O3DE), it's helpful to generate a dump file with crash reporting information to help track down the problem. 
 ---
 
+If you encounter a crash with **Open 3D Engine (O3DE)**, you may find helpful information in the dump file, `error.dmp`. The dump file records details about the state of the engine that led to the event of the crash. The dump file is a compact minidump and is located in the `user/log/` directory, within the same parent directory where you ran the O3DE Editor from. For example, if you ran the Editor from my project's directory, then you can find the dump files in `/MyProject/user/log/`.
```
Comment (byrcolin) [inline content/docs/user-guide/programming/debugging/crash-dumps.md:7]: For example, if you ran the Editor from the MyProject's directory, then you can find the dump files in `/MyProject/user/log/`.

File: content/docs/user-guide/programming/debugging/_index.md [hunk 3]
```diff
@@ -1,6 +1,17 @@
 ---
-title: "Debugging in O3DE"
-date: 2021-03-02T00:23:53-05:00
+title: Debugging in O3DE
+linktitle: Debugging
```
[the remark this answers is not in the document]
Comment (chanmosq) [reply content/docs/user-guide/programming/debugging/_index.md:3]: Created a new PR to resolve this

File: content/docs/user-guide/programming/debugging/crash-dumps.md (hunk 2 above)
Comment (chanmosq) [reply content/docs/user-guide/programming/debugging/crash-dumps.md:7]: Created a new PR to resolve this.

**Pull request merged**